### PR TITLE
feat: Support ESLint 7.x

### LINKED ISSUE #29: Support ESLint 7.x
ESLint v7.0.0 is [released](https://eslint.org/blog/2020/05/eslint-v7.0.0-released) 🎉

It would be awesome to have official ESLint 7 support. 👊
I'm happy to help where I can of course 🙂

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
ESLint v7.0.0 is [released](https://eslint.org/blog/2020/05/eslint-v7.0.0-released) 🎉

(dev)Dependencies should be compatible with ESLint 7 too before we can merge this one:

- [ ] [`@mysticatea/eslint-plugin`](https://github.com/mysticatea/eslint-plugin) (https://github.com/mysticatea/eslint-plugin/issues/28)
  - [ ] https://github.com/mysticatea/eslint-plugin/pull/29
  - [ ] Release

---

Closes #29

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
@mysticatea If you can point me into the right direction for what's wrong, I'm happy to make CI pass 🙂

**Comment by mysticatea** [comment]
Thank you for your contribution.

1. Please don't make breaking changes. Dropping old nodejs support is not required.
1. The values of `node.start` and `node.range[0]` can be different. In this case, the `node.start` of `TemplateElement` nodes is wrong -- it doesn't contain backtick. On the other hand, the `node.range[0]` of `TemplateElement` nodes is correct. You have to address it.

**Comment by MichaelDeBoey** [comment]
@mysticatea I've updated tests to pass CI.
Only 2 tests are still failing, but don't know what to do with those.

Hopefully you can help me into the right direction on this one 🙂

**Comment by mysticatea** [comment]
Hi, please don't change tests. See https://github.com/mysticatea/eslint-plugin-es/pull/30#issuecomment-626278683

**Comment by MichaelDeBoey** [comment]
@mysticatea I think I'm going into the right direction, but still have some failures.

**Comment by mysticatea** [comment]
The problem is, I guess, the range of `text` and the range of `node` are different. Using `sourceCode.getText(node)` at all may solve.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -28,37 +28,37 @@ jobs:
...
+          - eslint: 7
+            node: 12
             os: ubuntu-latest
-          - eslint: 6.x
-            node: 10.x
-            os: ubuntu-latest
-          - eslint: 6.x
-            node: 8.x
+          - eslint: 7
+            node: 10
```
Comment (mysticatea) [inline .github/workflows/ci.yml:47]: Don't delete Node 8.x job.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -28,37 +28,37 @@ jobs:
...
-          - eslint: 4.x
-            node: 14.x
+          - eslint: 4
+            node: 14
             os: ubuntu-latest
           # On the minimum supported ESLint/Node.js version
-          - eslint: 4.x
-            node: "8.0.0"
+          - eslint: 4
+            node: 8.10.0
```
Comment (mysticatea) [inline .github/workflows/ci.yml:61]: Please don't change the minimum supposed Node.js version.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -28,37 +28,37 @@ jobs:
 
     strategy:
       matrix:
-        eslint: [6.x]
-        node: [14.x]
+        eslint: [7]
+        node: [14]
```
Comment (mysticatea) [inline .github/workflows/ci.yml:32]: I think this change (deleting `.x`) is unrelated to confirming ESLint 7 support. It's better if each PR contains only related changes.

**Review by mysticatea — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (MichaelDeBoey) [reply .github/workflows/ci.yml:47]: ESLint requires Node@^10.12.x || >12.x

File: .github/workflows/ci.yml (hunk 2 above)
Comment (MichaelDeBoey) [reply .github/workflows/ci.yml:61]: `8.10.0` is the minimum that's inside `package.json`'s `engines`

File: .github/workflows/ci.yml (hunk 3 above)
Comment (MichaelDeBoey) [reply .github/workflows/ci.yml:32]: Will do that in a separate PR then

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mysticatea) [reply .github/workflows/ci.yml]: ESLint 6 works on Node.js 8.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mysticatea) [reply .github/workflows/ci.yml:61]: Oh, you're right. I'm sorry.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mysticatea) [reply .github/workflows/ci.yml]: Ah, sorry. We have job for the minimum versions, so this is OK.

**Review by mysticatea — APPROVED** [review]
LGTM, thank you very much!
Currently I'm on mobile, so I will merge this later.

**Comment by MichaelDeBoey** [comment]
Would be awesome to get this one merged 👊

Happy to help out on the other ones too 🙂

**Comment by MichaelDeBoey** [comment]
@mysticatea Any idea when this will be released? 🤔

**Pull request merged**